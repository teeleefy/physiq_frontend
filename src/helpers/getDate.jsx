function getDate() {
    const today = new Date();
    const year = today.getFullYear();
    let month = today.getMonth() + 1;
    if(month<10){
        month = `0${month}`
    }
    let date = today.getDate();
    if(date<10){
        date = `0${date}`
    }
    return `${year}-${month}-${date}`;
  }

export default getDate;
