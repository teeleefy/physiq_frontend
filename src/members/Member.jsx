import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import {useState, useEffect, useContext} from "react";
import { useNavigate } from "react-router-dom";
import PhysiqApi from "../Api.js";
import { NavLink } from "react-router-dom";
import MemberNavBar from "../navigation/MemberNavBar.jsx";
import Loading from "../navigation/Loading";
import './styles/Member.css'
import {MemberContext} from "../auth/UserContext";

function Member({updateMember, getDate}){
    const { currentMember } = useContext(MemberContext);
    let navigate = useNavigate();
    const [isLoading, setIsLoading] = useState(true);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({firstName: currentMember.firstName, lastName: currentMember.lastName, birthday: currentMember.birthday});
    const todaysDate = getDate();

    async function handleSubmit(evt){
        evt.preventDefault();
        let result = await updateMember(formData);
        if(result.success){
           setFormMessages(['Member Updated!'])
           setUpdateSuccess(true);
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };

    async function deleteMember() {
        try {
          let deletedMember= await PhysiqApi.deleteMember(currentMember.id);
          return { success: true };
        } catch (errors) {
          console.error("Delete failed", errors);
          return { success: false, errors };
        }
      }

    async function handleDelete(evt){
        evt.preventDefault();
        const shouldDelete = confirm("Are you sure you want to delete? THIS CANNOT BE UNDONE!!!")

        if(shouldDelete){
            let result = await deleteMember();
            if(result.success){
            //    alert("Saved Changes!") 
            setFormMessages(['Member Deleted!'])
            setUpdateSuccess(true);
            navigate("..", { relative: "path"});
            }
            else{
                setFormMessages(result.errors);
                setUpdateSuccess(false);
            }
        }
        
    };
  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };



    if(!currentMember) return <Loading />;
    return(
        <>
            <Form className="Member-Form m-4" >
            <h1 className="Member-h1">Update Member:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="Member-label" for="firstName" >
                    <b>First Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="firstName"
                name="firstName"
                value={formData.firstName}
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Member-label" for="lastName">
                    <b>Last Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="lastName"
                name="lastName"
                value={formData.lastName}
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup className="Member-date-input">
                <Label className="Member-label" for="birthday">
                    <b>Birthday</b>
                </Label>
                <Input
                id="birthday"
                name="birthday"
                value={formData.birthday}
                max={todaysDate}
                type="date"
                onChange={handleChange}
                />
                </FormGroup>

                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                {/* <Button className="btn-danger m-2" onClick={handleDelete}>DELETE MEMBER</Button> */}
                <Button className="btn-dark m-2" onClick={handleSubmit}>Save Changes</Button>
            </Form>
        </>
     )
}

export default Member;