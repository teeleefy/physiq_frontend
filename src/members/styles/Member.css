.Member-top-div {
    width: 90vw;
    height: 90vh;
    
    background-color: white;
    
    color: black;
    border: 15px red solid;
    border-radius: 15px;
    padding: 0;
    margin: 0;
  }

  .Member-card {
    height: 35vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Member-label {
    font-size: 1.2rem;
}

.Member-h1 {
    font-size: 2rem;
    font-weight: 600;
}

/* .Meds-navlink {
    text-decoration: none;
    color: white;
} */

.Member-date-input {
    width: 20vw;
}