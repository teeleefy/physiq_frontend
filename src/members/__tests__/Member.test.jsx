import {renderWithRouterAndMemberContext, screen} from '../../../__tests__/test-tools';
import getDate from "../../helpers/getDate.jsx"
import Member from '../Member';

describe('Member', () => {
    it('renders the Member component', () => {
      renderWithRouterAndMemberContext(<Member getDate={getDate}/>);
      screen.debug(); 
    });

    it("matches snapshot", function() {
      const { asFragment } = renderWithRouterAndMemberContext(<Member getDate={getDate}/>);
      expect(asFragment()).toMatchSnapshot();
    });

  });