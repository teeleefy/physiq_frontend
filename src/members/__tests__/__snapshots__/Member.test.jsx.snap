// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Member > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Member-Form m-4"
  >
    <h1
      class="Member-h1"
    >
      Update Member:
    </h1>
    <p
      class="text-secondary"
    >
      Fields marked with 
      <span
        class="text-danger"
      >
        *
      </span>
       are required.
    </p>
    <div
      class="mb-3"
    >
      <label
        class="Member-label form-label"
        for="firstName"
      >
        <b>
          First Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="firstName"
        maxlength="35"
        minlength="2"
        name="firstName"
        type="text"
        value="Sarah"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Member-label form-label"
        for="lastName"
      >
        <b>
          Last Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="lastName"
        maxlength="50"
        minlength="2"
        name="lastName"
        type="text"
        value="Smith"
      />
    </div>
    <div
      class="Member-date-input mb-3"
    >
      <label
        class="Member-label form-label"
        for="birthday"
      >
        <b>
          Birthday
        </b>
      </label>
      <input
        class="form-control"
        id="birthday"
        max="2024-08-05"
        name="birthday"
        type="date"
        value="1980-09-03"
      />
    </div>
    <button
      class="btn-danger m-2 btn btn-secondary"
      type="button"
    >
      DELETE MEMBER
    </button>
    <button
      class="btn-dark m-2 btn btn-secondary"
      type="button"
    >
      Save Changes
    </button>
  </form>
</DocumentFragment>
`;
